src/native/minipal/memory.c

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "memory.h"
+
+#ifdef HOST_WINDOWS
+#include <Windows.h>
+
+void* minipal_co_task_mem_alloc(size_t cb)
```

Comment:
Is there advantage in wrapping Windows concepts like CoTaskMemAlloc/Free in another layer vs. just implementing them on non-Windows using the Windows name? It is partial Windows emulator either way.

Reply:
Personally, I like having separate APIs and not re-using the Windows names. It provides a better IDE experience, especially when you don't also emulate all of the Windows defines and typedefs. I also like having our own names for general concepts, instead of just directly emulating Windows (although these few cases really are just wrapping Windows APIs, I admit).

Reply:
> Personally, I like having separate APIs and not re-using the Windows names.

Agree in general.

> I also like having our own names for general concepts, instead of just directly emulating Windows (although these few cases really are just wrapping Windows APIs, I admit).

I think this is the crux of the issue for me. We have `Marshal.AllocCoTaskMem()` and that is how I can see an argument for keeping `CoTaskMemAlloc()` as is and just implementing it in a PAL. If the name of the API was `Marshal.AllocComMemory()`, then I think abstracting it away is appropriate, but we are bound to some of the early COM API names and hiding them, in my mind, makes things less clear.

Reply:
Based on https://github.com/dotnet/runtime/pull/108999#discussion_r1813258952, I assume that we are not going to wrap everything to simulate the Windows COM environment. I think it would be more consistent to keep the Windows names for everything that has to do with simulating the Windows COM environment.

I agree with having our own names for general concepts as needed.

Reply:
I'll change the memory ones (and do a review of the other ones) to be named with the Windows names on non-Windows and not exist on Windows.